plugins/webp-uploads/tests/test-picture-element.php

```diff
@@ -50,8 +50,22 @@ public function test_maybe_wrap_images_in_picture_element( bool $jpeg_and_webp,
 			$this->assertStringNotContainsString( $picture_element, $the_image );
 		}
 
-		// When both features are enabled, the picture element will contain two srcset elements.
-		$this->assertEquals( ( $jpeg_and_webp && $expect_picture_element ) ? 2 : 1, substr_count( $the_image, 'srcset=' ) );
+		// When both features are enabled, the picture element will contain 3 srcset elements.
+		if ( $jpeg_and_webp && $expect_picture_element ) {
+			$this->assertEquals( 3, substr_count( $the_image, 'srcset=' ) );
+		}
+		// When neither features are enabled, the picture element will contain 1 srcset elements.
+		if ( ! $jpeg_and_webp && ! $expect_picture_element ) {
+			$this->assertEquals( 1, substr_count( $the_image, 'srcset=' ) );
+		}
+		// When only jpeg_and_webp is enabled, the picture element will contain 1 srcset elements.
+		if ( $jpeg_and_webp && ! $expect_picture_element ) {
+			$this->assertEquals( 1, substr_count( $the_image, 'srcset=' ) );
+		}
+		// When only picture_element is enabled, the picture element will contain 2 srcset elements.
+		if ( ! $jpeg_and_webp && $expect_picture_element ) {
+			$this->assertEquals( 2, substr_count( $the_image, 'srcset=' ) );
+		}
```

Comment:
Instead of `$expect_picture_element` I think it would perhaps be helpful if the actual HTML for the `picture` element is passed into this test form the `data_provider_it_should_maybe_wrap_images_in_picture_element` data provider. For example, it could be `$expected_html` and then this here can just do:

```
$this->assertEquals( $expected_html, $the_image );
```

By doing such snapshot comparisons, we'll be able to check for more expected markup than just the `srcset`, for example whether `alt` appears on the `img`.

Reply:
I was trying to implement this but the `src` and `srcset` keep on changing whenever we run the test case. Therefore, `$this->assertEquals( $expected_html, $the_image );` will may not work here.

@westonruter

Reply:
@devansh016 Since data providers are evaluated before test cases, and the generated attachments are dynamic, you can use placeholders for image src in the expected content like `{{img-src}}`, which can be search and replaced during the actual test.

Reply:
I have updated the tests.